doFillValidation.py

```diff
@@ -401,11 +406,17 @@ def getValidSections(fillNumber, l):
     os.system('brilcalc lumi -f '+str(fillNumber)+' --type '+l+' -b "STABLE BEAMS" --byls -o '+tempFileName)
     with open(tempFileName) as csv_input:
         reader = csv.reader(csv_input, delimiter=',')
+        lastrun=-1
         for row in reader:
             if row[0][0] == '#':
                 continue
             runfill=row[0].split(':')
             run=int(runfill[0])
+            # remove first LS of run (not necessarily LS=1)
```

Comment:
two questions:
1) is it really that the problem is with the first lumisection that DT has data for, even if that's not LS 1? that seems like a kind of weird behavior -- but I'll defer to your knowledge.
2) if you do the removal here, then all of the first lumisections will show up in the "missing" list for DT. I wanted to avoid this for BCM1F because I didn't want real problems to get buried in the list, which is why I did the removal in produceOutput() instead.

Reply:
1. It would be LS 1, except there are problems with the numbering.  You can see the offset in webmonitor.  So sometimes LS 2 is actually the first one... worse the offset is not the same all the time (can be up to 3 off).  I think it is best like this.
2.  It might as well be absent.  This is long standing problem.  We saw it in the 2015 and 2016 data too.  For all runs we'll have to remove the first LS.   (I think I had a technical problem with the other implementation, but I'm not married to any implementation.)

(Personally, I rather reject this LS in the processor because it is always bad, but I don't have time to do that out right now.)

Reply:
Ahh gotcha. OK, that makes sense.

On the second point, of course we'll have to remove it -- I'd just prefer to do it in produceOutput() because then you don't overwhelm the user with a bunch of warnings for every first LS. I think something like this would work fine:

    for r in sorted(recordedLumiSections.keys()):
        # look for the first LS for which DT is present
        for ls in sorted(recordedLumiSections[r].keys()):
            if 'dt' in recordedLumiSections[r][ls]:
                # once found, remove it
                recordedLumiSections[r][ls].remove(l)
                break

Reply:
OK.  I'm not sure how to check it now that I have gone through everything in my area...

Reply:
Do I just need to remove a fill from the list of check fills and re-run?

Reply:
Easiest way (or at least the way that I do it) is to just put something like
fillList = [6156]
after (or in place of) the command to populate fillList normally.

Reply:
Ok.  

Where would you put this code?

for r in sorted(recordedLumiSections.keys()):
    # look for the first LS for which DT is present
    for ls in sorted(recordedLumiSections[r].keys()):
        if 'dt' in recordedLumiSections[r][ls]:
            # once found, remove it
            recordedLumiSections[r][ls].remove('dt')
            break

Reply:
l. 491 (where the equivalent code for BCM1F used to live, as you can see in the comment).